bazel/compiler_id.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Given an argument that is the path to a compiler, tries to determine an
+# identification string for the compiler according to the CMake-defined IDs:
+# https://cmake.org/cmake/help/v3.5/variable/CMAKE_LANG_COMPILER_ID.html
+
+if [[ ! -e "${1}" ]]; then
+  echo "Usage: $0 <path-to-compiler>"
+  exit 1
+fi
+
+version="$("${1}" --version 2> /dev/null \
```

Comment:
If we're going to support multiple compilers, the script will probably grow larger as we detect the executable and use the appropriate switch/flags. Just spitballing, but what do you think about adopting an approach similar to [this](https://github.com/TristanThrush/drake-9660/blob/e8dd628593786a6ddfed6758bc4ab3c2d58ae0c4/tools/workspace/cc/identify_compiler.cc)? TL;DR, compile a program in Bazel that uses compiler-specific version macros then spits out the information.